File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/FileTagResultModel.java
package com.seafile.seadroid2.framework.model.sdoc;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.JsonAdapter;
import com.seafile.seadroid2.framework.model.adapter.OffsetDateTimeAdapter;

import java.util.List;

public class FileTagResultModel implements Parcelable {
    public String _creator;

    public String _id;
    public String _last_modifier;

    public String _tag_color;
    public String _tag_name;
    public List<FileLinkModel> _tag_file_links;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this._creator);
        dest.writeString(this._id);
        dest.writeString(this._last_modifier);
        dest.writeString(this._tag_color);
        dest.writeString(this._tag_name);
        dest.writeTypedList(this._tag_file_links);
    }

    public FileTagResultModel() {
    }

    protected FileTagResultModel(Parcel in) {
        this._creator = in.readString();
        this._id = in.readString();
        this._last_modifier = in.readString();
        this._tag_color = in.readString();
        this._tag_name = in.readString();
        this._tag_file_links = in.createTypedArrayList(FileLinkModel.CREATOR);
    }

    public static final Parcelable.Creator<FileTagResultModel> CREATOR = new Parcelable.Creator<FileTagResultModel>() {
        @Override
        public FileTagResultModel createFromParcel(Parcel source) {
            return new FileTagResultModel(source);
        }

        @Override
        public FileTagResultModel[] newArray(int size) {
            return new FileTagResultModel[size];
        }
    };
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/FileTagResultModel.kt
package com.seafile.seadroid2.framework.model.sdoc

import android.os.Parcel
import android.os.Parcelable

class FileTagResultModel() : Parcelable {
    @JvmField
    var _creator: String? = null

    @JvmField
    var _id: String? = null

    @JvmField
    var _last_modifier: String? = null

    @JvmField
    var _tag_color: String? = null

    @JvmField
    var _tag_name: String? = null

    @JvmField
    var _tag_file_links: MutableList<FileLinkModel>? = null

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(_creator)
        dest.writeString(_id)
        dest.writeString(_last_modifier)
        dest.writeString(_tag_color)
        dest.writeString(_tag_name)
        dest.writeTypedList(_tag_file_links)
    }

    private constructor(parcel: Parcel) : this() {
        _creator = parcel.readString()
        _id = parcel.readString()
        _last_modifier = parcel.readString()
        _tag_color = parcel.readString()
        _tag_name = parcel.readString()
        _tag_file_links = parcel.createTypedArrayList(FileLinkModel.CREATOR)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<FileTagResultModel> =
            object : Parcelable.Creator<FileTagResultModel> {
                override fun createFromParcel(source: Parcel): FileTagResultModel = FileTagResultModel(source)
                override fun newArray(size: Int): Array<FileTagResultModel?> = arrayOfNulls(size)
            }
    }
}
